pkg/getty/etc/profile.d/namespace.sh

```diff
@@ -1 +1,23 @@
-export PS1="(ns: getty) $PS1"
+#/bin/sh
+
+in_a_ns() {
+	IN_A_NS=0
+	NOT_IN_A_NS=1
+
+	# If PID 1 != /sbin/init then we are in a container
+	if ! pgrep /sbin/init > /dev/null; then
+		return $IN_A_NS
+	fi
+
+	# If containerd or runc are in the pstree then we're probably in a namespace
+	if pstree | grep -qE "(containerd|runc)"; then
+		return $IN_A_NS
+	fi
```

Comment:
> it will block forever since the children it is waiting for are actually the calls to infinit_loop which never exits

Doh! Of course it is! It has to be to function in the container! Ouch.